recipes/scc/all/conanfile.py

```diff
@@ -1,89 +1,176 @@
+            # and
+            #   src\sysc\tlm\scc\tlm_mm.h(31,20): error C2061: syntax error: identifier '__attribute__'
+            raise ConanInvalidConfiguration(f"{self.ref} is not supported on MSVC.")
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 11)
-        if self.settings.os == "Macos":
-            raise ConanInvalidConfiguration(f"{self.name} is not suppported on {self.settings.os}.")
+            check_min_cppstd(self, 11)
         if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "7":
             raise ConanInvalidConfiguration("GCC < version 7 is not supported")
 
-    def source(self):
-        get(self, **self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
     def build_requirements(self):
-        self.tool_requires("cmake/3.24.0")
+        self.tool_requires("cmake/[>=3.24 <4]")
 
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["SC_WITH_PHASE_CALLBACKS"] = self.options.enable_phase_callbacks
-        cmake.definitions["SC_WITH_PHASE_CALLBACK_TRACING"] = self.options.enable_phase_callbacks_tracing
-        cmake.definitions["BUILD_SCC_DOCUMENTATION"] = False
-        cmake.definitions["SCC_LIB_ONLY"] = True
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["SC_WITH_PHASE_CALLBACKS"] = self.options.enable_phase_callbacks
+        tc.cache_variables["SC_WITH_PHASE_CALLBACK_TRACING"] = self.options.enable_phase_callbacks_tracing
+        tc.cache_variables["BUILD_SCC_DOCUMENTATION"] = False
+        tc.cache_variables["SCC_LIB_ONLY"] = True
+        tc.cache_variables["ENABLE_CONAN"] = False
         if self.settings.os == "Windows":
-            cmake.definitions["SCC_LIMIT_TRACE_TYPE_LIST"] = True
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+            tc.cache_variables["SCC_LIMIT_TRACE_TYPE_LIST"] = True
+        # Used at https://github.com/Minres/SystemC-Components/blob/2023.06/src/common/util/pool_allocator.h#L110
+        # but is not set anywhere
+        tc.preprocessor_definitions["_GLIBCXX_USE_NOEXCEPT"] = "noexcept"
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.set_property("systemc", "cmake_file_name", "SystemC")
+        deps.set_property("systemc-cci", "cmake_target_name", "systemc-cci::systemc-cci")
+        deps.set_property("yaml-cpp", "cmake_target_name", "yaml-cpp::yaml-cpp")
+        deps.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        replace_in_file(self, os.path.join(self.source_folder, "third_party", "axi_chi", "CMakeLists.txt"),
+                        " STATIC", "")
+        if is_msvc(self):
```

Comment:
This is explicitly forbidden in ``validate()`` isn't it? Maybe can be removed?

Reply:
Removed them.